viadot/sources/supermetrics.py

```diff
@@ -0,0 +1,164 @@
+                "Credentials not specified. Falling back to `boto3` default credentials."
+            )
+
+        self.query_params = query_params
+
+    @classmethod
+    def get_params_from_api_query(cls, url: str) -> Dict[str, Any]:
+        """Returns parmeters from API query in a dictionary"""
+        url_unquoted = urllib.parse.unquote(url)
+        s = urllib.parse.parse_qs(url_unquoted)
+        endpoint = list(s.keys())[0]
+        params = s[endpoint][0]
+        params_d = json.loads(params)
+        return params_d
+
+    @classmethod
+    def from_url(cls, url: str, credentials: Dict[str, Any] = None):
+        obj = Supermetrics(
+            credentials=credentials or get_source_credentials("supermetrics")
+        )
+        params = cls.get_params_from_api_query(url)
+        obj.query_params = params
+        return obj
+
+    def to_json(self, timeout=(3.05, 60 * 30)) -> Dict[str, Any]:
+        """Download query results to a dictionary.
+        Note that Supermetrics API will sometimes hang and not return any error message,
+        so we're adding a timeout to GET.
+
+        See [requests docs](https://docs.python-requests.org/en/master/user/advanced/#timeouts)
+        for an explanation of why this timeout value will work on long-running queries but fail fast
+        on connection issues.
+        """
+
+        if not self.query_params:
+            raise ValueError("Please build the query first")
+
+        params = {"json": json.dumps(self.query_params)}
+        headers = {"Authorization": f'Bearer {self.credentials.get("api_key")}'}
+
+        response = handle_api_response(
+            url=self.API_ENDPOINT, params=params, headers=headers, timeout=timeout
+        )
+        return response.json()
+
+    @classmethod
+    def _get_col_names_google_analytics(
+        cls,
+        response: dict,
+    ) -> List[str]:
+        """Returns list of Google Analytics columns names"""
+
+        # Supermetrics allows pivoting GA data, in which case it generates additional columns,
+        # which are not enlisted in response's query metadata but are instead added as the first row of data.
+        is_pivoted = any(
+            field["field_split"] == "column"
+            for field in response["meta"]["query"]["fields"]
+        )
+
+        if is_pivoted:
+            if not response["data"]:
+                raise ValueError(
+                    "Couldn't find column names as query returned no data."
+                )
+            columns = response["data"][0]
+        else:
+            # non-pivoted data; query fields match result fields
+            cols_meta = response["meta"]["query"]["fields"]
+            columns = [col_meta["field_name"] for col_meta in cols_meta]
+        return columns
+
+    @classmethod
+    def _get_col_names_other(cls, response: dict) -> List[str]:
+        """Returns list of columns names (to Google Analytics use  _get_col_names_google_analytics ()"""
```

Comment:
I think we can standardize docstrings as in other sources https://github.com/dyvenia/viadot/blob/9239fcbf185f93230407fd16b4e96cd081e3d09b/viadot/sources/redshift_spectrum.py#L109